src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    if (this[0] !== null && this[0] !== undefined) {
+      resultString = `${this[0]}`;
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
From CHECKLIST:

> [CODE STYLE] - avoid using nested `if` statements, we are sure that you can do it without them :)

Reply:
Here it's okay. It's not a nested `if statement`. Example of nested if:
`if (...) {
  if(...) {
    if(...) {
      ...
    }
  }
}`